framework/include/parser/CommandLine.h

```diff
@@ -191,3 +191,13 @@ CommandLine::search(const std::string & option_name, T & argument)
   }
   mooseError("Unrecognized option name");
 }
+
+enum HitCmpResult
+{
+	GLOBAL,
+	MAIN,
+	MATCH,
+	WRONG
+};
+
+HitCmpResult hitArgCmp(std::string arg, std::string appname, int appnum);
```

Comment:
In general you want to pass more expensive objects by const reference in C++.  (e.g. `const std::string &`)

Reply:
This is more of a MOOSE thing, but we almost always prefer unsigned types versus signed types. It's rare to see `int` throughout MOOSE (compared to `unsigned int`) although I realize it's very commonplace in "C". The reason for this is that you often due comparisons with things like indices for standard containers, which are all unsigned. This yields a compiler warning, which we don't allow.

Reply:
One more thing on style. In C++ we have very few free functions that are used outside of the current compilation unit. This "utility" function would normally be placed inside of an anonymous namespace so it would be essentially private to everything outside of CommandLine.C. However, the whole point of this PR is to utilize this utility from multiple classes, which you have done. You have a few options that would match the style in our project a little better. 

1) Place this function in one of our utility files like `MooseUtil.C` which is enclosed in a namespace. Then people who see this in the code understand where to look for the implementation because of the namespace you have to use when calling the method (e.g. `MooseUtils::hitArgCmp()`)

2) Make it a static function on a class where it fits best organizationally. Since this is for comparing command line arguments. It fits reasonably well as a static method on CommandLine. This essentially gives you the same calling semantics. Now to call the function I would say `CommandLine::hitArgCmp()` 

One last thought - Generally we are verbose and tend to avoid abbreviating things too much. Some of us might name this method `hitArgumentComparison` or `hitArgComparison`. We tend to think it makes more readable code especially when more jargon words are involved in the names. "Arg" and "Cmp" are arguably very common so I'm not going to be super critical about this name. Take a look at MooseUtil.C and see the typical function names in there.